pkg/ctxutil/ctxutil.go

```diff
@@ -489,3 +490,18 @@ func IsHidden(ctx context.Context) bool {
 
 	return bv
 }
+
+// WithAutoSync returns a context with the autosync flag set.
+func WithAutoSync(ctx context.Context, autosync bool) context.Context {
+	return context.WithValue(ctx, ctxKeyAutoSync, autosync)
+}
+
+// HasAutoSync returns true if syncs are triggered by autosync.
+func HasAutoSync(ctx context.Context) bool {
+	bv, ok := ctx.Value(ctxKeyAutoSync).(bool)
+	if !ok {
+		return false
+	}
+
+	return bv
+}
```

Comment:
Do you really need these?
It looks like you're relying on the interactive context key above, instead of that newly added autosync one, no?